gc/2022/GC-02-22.md

```diff
@@ -59,11 +59,11 @@ TL: Binaryen & V8 have been experimenting with versions of the instructions that
 
 TL: There seem to be two arguments for RTTs: JS API and generics. For the JS API, I'd suggest we think about that without RTTs being a hard requirement.
 
-AR: I see one more reason, beyond generics & JS API (which I'd put last): having an explicit cost model for Wasm. With the MVP proposal, there was meant to be a 1:1 relationship between Wasm instructions and machine instructions; this was what was "low level" about Wasm. With GC we have to expand this to maintain safety [not 1:1 anymore]. Still important that producers be able to do optimizations without depending on sophisticated VM optimizations. It's all a bit fuzzy, but where we can, we should make costs explicit where it makes sense. RTTs try to do that: they represent operations that would be separate if producers were lowering to machine code instead of Wasm. With generics this becomes even more important. This was the original motivation for RTTs to exist. The JS API use case followed once RTTs existed.
+AR: I see one more reason, beyond generics & JS API (which I'd put last): having an explicit cost model for Wasm. With the MVP proposal, there was meant to be a 1:1 relationship between Wasm instructions and machine instructions; this was what was "low level" about Wasm. With GC we have to relax this to maintain safety, but should still stick as close to the ideal as possible. Still important that producers be able to do optimizations without depending on sophisticated VM optimizations. It's all a bit fuzzy, but where we can, we should make costs explicit where it makes sense. RTTs try to do that: they represent operations that would be separate if producers were lowering to machine code instead of Wasm. With generics this becomes even more important. This was the original motivation for RTTs to exist. The JS API use case followed once RTTs existed.
 
 BT: Cost argument is a sliding scale. If we applied it universally, e.g. call_indirect would be split into multiple operations. My reason for wanting to keep RTTs is that if they are implicit, that might be an extra load from the instance object. But that depends on implementation details, so the explicit cost argument cuts both ways.
 
-AR: Fair, it is a blurry line. My argument is that, if it's something we'll need anyway, we should add it now since it also fits with the cost model. call_indirect is an interesting example, I don't think anyone was really happy with how that came out; if we had to do it again I could imagine it being designed differently.
+AR: Fair, it is a blurry line. My argument is that, if it's something we'll need anyway, we should add it now since it also fits with the cost model. call_indirect is an interesting example, I don't think anyone was really happy with how that came out; if we had to do introduce it at this point I could imagine it being designed differently, e.g., in terms of casts.
```

Comment:
```suggestion
AR: Fair, it is a blurry line. My argument is that, if it's something we'll need anyway, we should add it now since it also fits with the cost model. call_indirect is an interesting example, I don't think anyone was really happy with how that came out; if we had to introduce it at this point I could imagine it being designed differently, e.g., in terms of casts.
```